### PR TITLE
[9.12.r1] staging: android: lowmemorykiller: Port fixes

### PR DESCRIPTION
**Pull request by Thaodan**
A few smaller fixes needed to make the Android LowMemoyKiller properly build with 4.19.

### CODE REVIEW

File: drivers/staging/android/lowmemorykiller_tng.c [hunk 1]
```diff
@@ -161,8 +161,8 @@ static void calc_params(struct calculated_params *cp, gfp_t mask)
 	int i;
 	int array_size;
 	long free_swap = get_nr_swap_pages();
-	long irp = global_node_page_state(NR_INDIRECTLY_RECLAIMABLE_BYTES)
-	  >> PAGE_SHIFT;
+	long irp = global_node_page_state(NR_SLAB_RECLAIMABLE) +
+            global_node_page_state(NR_KERNEL_MISC_RECLAIMABLE);
```
Comment (sledges) [inline drivers/staging/android/lowmemorykiller_tng.c]: Use tabs instead of spaces.

File: include/linux/mmzone.h [hunk 2]
```diff
@@ -778,6 +778,7 @@ static inline bool pgdat_is_empty(pg_data_t *pgdat)
 void build_all_zonelists(pg_data_t *pgdat);
 #ifdef CONFIG_ANDROID_LOW_MEMORY_KILLER_CONSIDER_SWAP
 void _wake_all_kswapds(unsigned int order,
+                             gfp_t gfp_mask,
```
Comment (sledges) [inline include/linux/mmzone.h]: Same as above.

File: mm/page_alloc.c [hunk 3]
```diff
@@ -4294,6 +4294,7 @@ static void wake_all_kswapds(unsigned int order, gfp_t gfp_mask,
 
 #ifdef CONFIG_ANDROID_LOW_MEMORY_KILLER_CONSIDER_SWAP
 void _wake_all_kswapds(unsigned int order,
+                             gfp_t gfp_mask,
```
Comment (sledges) [inline mm/page_alloc.c]: Same as above.

**Review by sledges — CHANGES_REQUESTED** [review]

**Comment by sledges** [comment]
Typo: should be `gfp_mask` in `staging: android: lowmemorykiller: Use gfp_mak in consider swap`.

File: drivers/staging/android/lowmemorykiller_tng.c (hunk 1 above)
Comment (sledges) [inline drivers/staging/android/lowmemorykiller_tng.c]: Here you are using this conversion: https://github.com/sonyxperiadev/kernel/commit/b29940c1abd7a4c3abeb926df0a5ec84d6902d47#diff-cef95765dfd76e5f9c9f0faebfa683edf904d0c3de71547ae8c3ea14418c1e38R4734

File: drivers/staging/android/lowmemorykiller_tng.c [hunk 4]
```diff
@@ -302,7 +302,7 @@ void print_obituary(struct task_struct *doomed,
 		     global_node_page_state(NR_SLAB_UNRECLAIMABLE) *
 		     (long)(PAGE_SIZE / 1024),
 		     gfp_mask,
-		     global_node_page_state(NR_INDIRECTLY_RECLAIMABLE_BYTES)
+		     global_node_page_state(NR_KERNEL_MISC_RECLAIMABLE)
```
Comment (sledges) [inline drivers/staging/android/lowmemorykiller_tng.c]: and here this one: https://github.com/sonyxperiadev/kernel/commit/b29940c1abd7a4c3abeb926df0a5ec84d6902d47#diff-fa19cdd9c3369d7f59aa2e8404628109408dbf8e1b568d1157a27328f75b8410R681

The `		     "   Indirect Reclaimable is %zdkB\n"` above should also be renamed accordingly I believe.

**Review by sledges — CHANGES_REQUESTED** [review]

**Comment by sledges** [comment]
> A few smaller fixes needed to make the Android LowMemoRyKiller properly build with 4.19.

lowmemorykiller builds fine as it is.
These fixes are for when CONFIG_ANDROID_LOW_MEMORY_KILLER_TNG and CONFIG_ANDROID_LOW_MEMORY_KILLER_CONSIDER_SWAP are enabled.

File: drivers/staging/android/lowmemorykiller_tng.c (hunk 4 above)
Comment (Thaodan) [reply drivers/staging/android/lowmemorykiller_tng.c]: Adjusted.

File: drivers/staging/android/lowmemorykiller_tng.c [hunk 5]
```diff
@@ -302,7 +302,7 @@ void print_obituary(struct task_struct *doomed,
...
 		     / 1024,
```
Comment (sledges) [inline drivers/staging/android/lowmemorykiller_tng.c]: `NR_KERNEL_MISC_RECLAIMABLE` is in pages, not bytes (as per b29940c1abd7a4c3abeb926df0a5ec84d6902d47)

File: drivers/staging/android/lowmemorykiller_tng.c [hunk 6]
```diff
@@ -161,8 +161,8 @@ static void calc_params(struct calculated_params *cp, gfp_t mask)
 	int i;
 	int array_size;
 	long free_swap = get_nr_swap_pages();
-	long irp = global_node_page_state(NR_INDIRECTLY_RECLAIMABLE_BYTES)
-	  >> PAGE_SHIFT;
+	long irp = global_node_page_state(NR_SLAB_RECLAIMABLE) +
+	    global_node_page_state(NR_KERNEL_MISC_RECLAIMABLE);
```
Comment (sledges) [inline drivers/staging/android/lowmemorykiller_tng.c]: I'll rephrase my previous comment: why are you also adding NR_SLAB_RECLAIMABLE here? (you omit it in `print_obituary(...)`)

Comment (Thaodan) [reply drivers/staging/android/lowmemorykiller_tng.c]: I replicated the changes done in b29940c1abd7a4c3abeb926df0a5ec84d6902d47.
In the commit NR_INDIRECTLY_RECLAIMABLE_BYTES was replaced with NR_SLAB_RECLAIMABLE +NR_KERNEL_MISC_RECLAIMABLE.

Comment (sledges) [reply drivers/staging/android/lowmemorykiller_tng.c]: It's also replaced without slab in the same commit here: https://github.com/sonyxperiadev/kernel/commit/b29940c1abd7a4c3abeb926df0a5ec84d6902d47?diff=unified#diff-fa19cdd9c3369d7f59aa2e8404628109408dbf8e1b568d1157a27328f75b8410R681

Comment (Thaodan) [reply drivers/staging/android/lowmemorykiller_tng.c]: Makes sense do it in the same way, adjusted.

File: drivers/staging/android/lowmemorykiller_tng.c (hunk 5 above)
Comment (Thaodan) [reply drivers/staging/android/lowmemorykiller_tng.c]: Converted from pages to memory as with other entries, adjusted.

File: drivers/staging/android/lowmemorykiller_tng.c (hunk 6 above)
Comment (sledges) [reply drivers/staging/android/lowmemorykiller_tng.c]: I've tested both approaches, the latter makes lmk to be less aggressive. Approving _post-mortem_:)

For future ref: variable `ipr` means Indirectly Reclaimable Pages, which should've been renamed in the light of ported changes.

Comment (Thaodan) [reply drivers/staging/android/lowmemorykiller_tng.c]: > For future ref: variable `ipr` means Indirectly Reclaimable Pages, which should've been renamed in the light of ported changes.


I was about to do that but the PR was already merged.

Comment (jerpelea) [reply drivers/staging/android/lowmemorykiller_tng.c]: @Thaodan open a new PR

**Pull request merged**